### PR TITLE
grass: fix build on aarch64-darwin

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

~This PR depends on #195162 and #195216: merging it alone will break x86_64-darwin support.~

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by willcohen — APPROVED** [review]

File: pkgs/applications/gis/grass/default.nix [hunk 1]
```diff
@@ -17,13 +17,11 @@ stdenv.mkDerivation rec {
...
   nativeBuildInputs = [ pkg-config makeWrapper ];
   buildInputs = [ flex bison zlib proj gdal libtiff libpng fftw sqlite
-  readline ffmpeg netcdf geos postgresql libmysqlclient blas
-  libLAS proj-datumgrid zstd wrapGAppsHook ]
-    ++ lib.optionals stdenv.isLinux [ cairo pdal wxGTK31 ]
-    ++ lib.optional stdenv.isDarwin wxmac
-    ++ (with python3Packages; [ python python-dateutil numpy ]
-      ++ lib.optional stdenv.isDarwin wxPython_4_0
-      ++ lib.optional stdenv.isLinux wxPython_4_1);
+  readline ffmpeg netcdf geos postgresql libmysqlclient blas wxGTK32
```
Comment (SuperSandro2000) [inline pkgs/applications/gis/grass/default.nix:20]: Formatting is of here

File: pkgs/applications/gis/grass/default.nix [hunk 2]
```diff
@@ -17,13 +17,11 @@ stdenv.mkDerivation rec {
...
+  proj-datumgrid zstd wrapGAppsHook ]
+    ++ lib.optionals stdenv.isLinux [ cairo pdal ]
+    ++ lib.optionals stdenv.isDarwin [ libiconv ]
+    ++ (with python3Packages; [ python python-dateutil numpy wxPython_4_1 ]);
```
Comment (SuperSandro2000) [inline pkgs/applications/gis/grass/default.nix:24]:
```suggestion
    ++ (with python3Packages; [ python-dateutil numpy wxPython_4_1 ]);
```

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -26624,7 +26624,9 @@ with pkgs;
 
   dcw-gmt = callPackage ../applications/gis/gmt/dcw.nix { };
 
-  grass = callPackage ../applications/gis/grass { };
+  grass = callPackage ../applications/gis/grass {
+    inherit (darwin) libiconv;
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:26628]: We don't need that inherit here

**Review by willcohen — APPROVED** [review]

File: pkgs/applications/gis/grass/default.nix [hunk 4]
```diff
@@ -16,14 +16,13 @@ stdenv.mkDerivation rec {
...
-  buildInputs = [ flex bison zlib proj gdal libtiff libpng fftw sqlite
-  readline ffmpeg netcdf geos postgresql libmysqlclient blas
-  libLAS proj-datumgrid zstd wrapGAppsHook ]
-    ++ lib.optionals stdenv.isLinux [ cairo pdal wxGTK31 ]
-    ++ lib.optional stdenv.isDarwin wxmac
-    ++ (with python3Packages; [ python python-dateutil numpy ]
-      ++ lib.optional stdenv.isDarwin wxPython_4_0
-      ++ lib.optional stdenv.isLinux wxPython_4_1);
+  buildInputs = [
+    cairo flex bison zlib proj gdal libtiff libpng fftw sqlite
```
Comment (SuperSandro2000) [inline pkgs/applications/gis/grass/default.nix:20]:
```suggestion
    cairo zlib proj gdal libtiff libpng fftw sqlite
```
Those go to nativeBuildInputs. Can we set ``strictDeps = true``?

File: pkgs/applications/gis/grass/default.nix [hunk 5]
```diff
@@ -16,14 +16,13 @@ stdenv.mkDerivation rec {
...
+    readline ffmpeg netcdf geos pdal postgresql libmysqlclient blas wxGTK32
+    proj-datumgrid zstd wrapGAppsHook
```
Comment (SuperSandro2000) [inline pkgs/applications/gis/grass/default.nix:22]:
```suggestion
    proj-datumgrid zstd
```
same as above

File: pkgs/applications/gis/grass/default.nix (hunk 4 above)
Comment (wegank) [reply pkgs/applications/gis/grass/default.nix]: Done.

**Pull request merged**